testutil/network/gatewaynet/network.go

```diff
@@ -0,0 +1,190 @@
+	memnet.CreateKeyringAccounts(t)
+
+	// Configure gateway and application module genesis states.
+	memnet.configureGatewayModuleGenesisState(t)
+	memnet.configureAppModuleGenesisState(t)
+
+	memnet.Network = network.New(t, *memnet.GetCosmosNetworkConfig(t))
+
+	memnet.FundOnChainAccounts(t)
+}
+
+// configureGatewayModuleGenesisState generates and populates the in-memory network's
+//
+//	application module's GenesisState object with the number of applications specified
+//	by the InMemoryConfig, each of which is staked for a unique service. It returns
+//	the genesis state object.
+func (memnet *inMemoryNetworkWithGateways) configureGatewayModuleGenesisState(t *testing.T) {
+	t.Helper()
+
+	gatewayGenesisState := gatewaytypes.DefaultGenesis()
+	for gatewayIdx := 0; gatewayIdx < memnet.Config.NumGateways; gatewayIdx++ {
+		stake := sdk.NewCoin("upokt", sdk.NewInt(10000))
+		preGeneratedAcct, ok := memnet.PreGeneratedAccountIterator.Next()
+		require.Truef(t, ok, "pre-generated accounts iterator exhausted")
+		require.Truef(t, ok, "pre-generated accounts iterator exhausted")
+
+		gateway := gatewaytypes.Gateway{
+			Address: preGeneratedAcct.Address.String(),
+			Stake:   &stake,
+		}
+
+		gatewayGenesisState.GatewayList = append(gatewayGenesisState.GatewayList, gateway)
+	}
+
+	gatewayGenesisBuffer, err := memnet.GetCosmosNetworkConfig(t).Codec.MarshalJSON(gatewayGenesisState)
+	require.NoError(t, err)
+
+	// Add supplier module genesis supplierGenesisState to the network config.
+	memnet.GetCosmosNetworkConfig(t).GenesisState[gatewaytypes.ModuleName] = gatewayGenesisBuffer
+}
+
+func (memnet *inMemoryNetworkWithGateways) configureAppModuleGenesisState(t *testing.T) {
+	t.Helper()
+
+	require.NotEmptyf(t, memnet.GetCosmosNetworkConfig(t), "cosmos config not initialized, call #Start() first")
+	require.NotEmptyf(t, memnet.PreGeneratedAccountIterator, "pre-generated accounts not initialized, call #Start() first")
+
+	var appGenesisState = apptypes.DefaultGenesis()
+	for appIdx := 0; appIdx < memnet.Config.GetNumApplications(t); appIdx++ {
+		preGeneratedAcct, ok := memnet.PreGeneratedAccountIterator.Next()
+		require.Truef(t, ok, "pre-generated accounts iterator exhausted")
+
+		application := apptypes.Application{
+			Address: preGeneratedAcct.Address.String(),
+			Stake:   &sdk.Coin{Denom: "upokt", Amount: sdk.NewInt(10000)},
+			ServiceConfigs: []*sharedtypes.ApplicationServiceConfig{
+				{Service: &sharedtypes.Service{Id: fmt.Sprintf("svc%d", appIdx)}},
+				// NB: applications are staked for a service which no supplier is staked for.
+				{Service: &sharedtypes.Service{Id: fmt.Sprintf("nosvc%d", appIdx)}},
+			},
+		}
+		appGenesisState.ApplicationList = append(appGenesisState.ApplicationList, application)
+	}
+	appGenesisBuffer, err := memnet.Config.CosmosCfg.Codec.MarshalJSON(appGenesisState)
+	require.NoError(t, err)
+
+	// Add supplier and application module genesis appGenesisState to the network memnetConfig.
```

Comment:
Isn't this only adding the application genesis state and not the supplier?

Reply:
Nice catch! :pray: